internal/backend/triggers/triggers.go

```diff
@@ -28,20 +31,75 @@ func (m *triggers) Create(ctx context.Context, trigger sdktypes.Trigger) (sdktyp
 
 	trigger = trigger.WithNewID()
 
+	if trigger.ConnectionID() == fixtures.BuiltinSchedulerConnectionID {
+		if schedule, _ := trigger.Data()[sdktypes.ScheduleExpression].ToString(); schedule != "" {
+			scheduleID := newScheduleID(trigger)
+			if err := m.scheduler.Create(ctx, scheduleID, schedule, trigger.ID()); err != nil {
+				return sdktypes.InvalidTriggerID, err
+			}
+			trigger = trigger.WithUpdatedData(sdktypes.ScheduleIDKey, sdktypes.NewStringValue(scheduleID))
+		}
+	}
+
 	if err := m.db.CreateTrigger(ctx, trigger); err != nil {
 		return sdktypes.InvalidTriggerID, err
 	}
-
 	return trigger.ID(), nil
 }
 
 func (m *triggers) Update(ctx context.Context, trigger sdktypes.Trigger) error {
+	prevTrigger, err := m.db.GetTrigger(ctx, trigger.ID())
+	if err != nil {
+		return err
+	}
+
+	prevData := prevTrigger.Data()
+	prevScheduleVal, isSchedulerPrevTrigger := prevData[sdktypes.ScheduleExpression]
+	scheduleID, _ := prevData[sdktypes.ScheduleIDKey].ToString()
+	isSchedulerPrevTrigger = isSchedulerPrevTrigger && prevTrigger.ConnectionID() == fixtures.BuiltinSchedulerConnectionID
+
+	data := trigger.Data()
+	scheduleVal, isSchedulerTrigger := data[sdktypes.ScheduleExpression]
+	schedule, _ := scheduleVal.ToString()
+	isSchedulerTrigger = isSchedulerTrigger && trigger.ConnectionID() == fixtures.BuiltinSchedulerConnectionID
+
+	if isSchedulerPrevTrigger {
```

Comment:
we need to figure out in a separate PR what happens if we crush here

Reply:
if we crash where? the comment is on `if` so I'm not sure that I understand

Reply:
what will happen if we crash in the middle of creating trigger? unrelated to schedule part..

Reply:
the scheduler makes this non-atomic, hence it is related

Reply:
my question is how we preserve atomicity on any other operation, e.g. creating event + event record. Ak could fail after creating the record but before creating the record.
the ak just between workflow creation and saving the trigger, this will mean that we have orphaned workflow that is running without the trigger. So we could ensure this in workflow code and if there is no trigger to remove the workflow. Right now workflow will be triggered by temporal each tick, but workflow function will fail since no such trigger will be found

Reply:
Added a Note in the code